core/src/main/java/hudson/diagnosis/TooManyJobsButNoView.java

```diff
@@ -67,5 +69,10 @@ public void doAct(StaplerRequest req, StaplerResponse rsp) throws IOException {
         }
     }
 
+    @Override
+    public Permission getRequiredPermission() {
+        return Jenkins.SYSTEM_READ;
```

Comment:
Unsure about that one, since SYSTEM_READ doesn't grant access to view jobs, and `getItemMap` bypasses regular Read permission checks. So you'll be told there's stuff there but cannot see it, which seems weird.

Reply:
I took the assumption that it’s highly unlikely that you wouldn’t be able to see jobs and it’s more of a quirk of the permissions system that it doesn’t automatically grant you read access,

I did hide the new view link as in read only systems it’s more likely views are added through a different process than directly in the UI

Reply:
Would make more sense if you accessed items (and views) through a method performing permission checks. That way, the result in the common case is identical, and in the unusual case you do not expect it's consistent with what the current user us seeing.

This should do it, without losing the performance benefit for admins:

```
        Jenkins j = Jenkins.get();
        if (j.hasPermission(Jenkins.ADMINISTER)) {
            return j.getViews().size()==1 && j.getItemMap().size()> THRESHOLD;
        }
        // SystemRead
        return j.getViews().size() == 1 && j.getItems().size() > THRESHOLD;
```

This + a quick amendment to `TooManyJobsButNoViewTest` and this is ready to go 👍

Reply:
> I took the assumption that it’s highly unlikely that you wouldn’t be able to see jobs and it’s more of a quirk of the permissions system that it doesn’t automatically grant you read access,

Good enough from my point of view

> Would make more sense if you accessed items (and views) through a method performing permission checks.

Could be a good improvement I would be happy to see Lazy loading for jobs eventually, and the implementation would require a change after that. But it would be the least of the problems we would have